test/NJsonApiCore.Test/Serialization/JsonApiTransformerTest/TestLinkData.cs

```diff
@@ -0,0 +1,119 @@
+﻿using NJsonApi;
+using NJsonApi.Serialization.Documents;
+using NJsonApi.Serialization.Representations;
+using NJsonApi.Serialization.Representations.Resources;
+using NJsonApi.Test.Builders;
+using NJsonApi.Test.TestControllers;
+using System;
+using Xunit;
+using NJsonApi.Infrastructure;
+
+namespace NJsonApi.Test.Serialization.JsonApiTransformerTest
+{
+    public class TestLinkData
+    {
+        [Fact]
+        public void Creates_CompoundDocument_for_single_class_with_no_links_and_properly_map_no_links()
+        {
+            // Arrange
+            var context = CreateContext();
+            SampleClass objectToTransform = CreateObjectToTransform();
+            var transformer = new JsonApiTransformerBuilder()
+                .With(CreateConfiguration())
+                .Build();
+
+            // Act
+            CompoundDocument result = transformer.Transform(objectToTransform, context);
+
+            // Assert
+            Assert.NotNull(result.Data);
+            var transformedObject = result.Data as SingleResource;
+            Assert.NotNull(transformedObject);
+        }
+
+        [Fact]
+        public void Creates_CompoundDocument_for_single_class_with_links_and_properly_map_links()
+        {
+            // Arrange
+            var context = CreateContext();
+            SampleClassWithObjectLinks objectToTransform = CreateObjectWithLinkDataToTransform();
+            var transformer = new JsonApiTransformerBuilder()
+                .With(CreateConfiguration())
+                .Build();
+
+            // Act
+            CompoundDocument result = transformer.Transform(objectToTransform, context);
+
+            // Assert
+            var transformedObject = result.Data as SingleResource;
+            Assert.Equal("url1", ((ISimpleLink)transformedObject.Links["link1"]).Href);
```

Comment:
Could we test the transformation of LinkObject too? In the same test or in a separate test?